File: js/web/examples/stream-events.mjs
import {
  Client,
  Filter,
  Kind,
  LogLevel,
  RelayUrl,
  ReqTarget,
  initLogger,
} from "../dist/index.js";

export async function streamEventsExample() {
  initLogger(LogLevel.Info);

  const client = new Client();
  await client.addRelay(RelayUrl.parse("wss://relay.damus.io"));
  await client.addRelay(RelayUrl.parse("wss://nos.lol"));
  await client.connect();

  const filter = new Filter().kind(new Kind(0)).limit(5n);
  const stream = await client.streamEvents(ReqTarget.auto([filter]), undefined, 10_000);
  while (true) {
    const item = await stream.next();
    if (item === undefined) break;
    if (item.event !== undefined) {
      console.log(item.event.asJson());
    } else if (item.error !== undefined) {
      console.log(`Relay error from ${item.relayUrl}: ${item.error}`);
    }
  }
}

File: android/examples/stream_events.kt
package org.nostrdevkit.examples

import org.nostrdevkit.sdk.*

suspend fun streamEventsExample() {
    initLogger(LogLevel.INFO)

    val client = Client()
    client.addRelay(RelayUrl.parse("wss://relay.damus.io"))
    client.addRelay(RelayUrl.parse("wss://nos.lol"))
    client.connect()

    println("Streaming events from relays...")
    val filter = Filter().kind(Kind(0u)).limit(5uL)
    val stream = client.streamEvents(ReqTarget.auto(listOf(filter)))
    while (true) {
        val item = stream.next() ?: break
        when {
            item.event != null -> println(item.event?.asJson())
            item.error != null -> println("Relay error from ${item.relayUrl}: ${item.error}")
        }
    }
    println("Stream terminated.")
}
